subsys/bluetooth/audio/has.c

```diff
@@ -32,8 +32,33 @@ LOG_MODULE_REGISTER(bt_has, CONFIG_BT_HAS_LOG_LEVEL);
  */
 #define BT_HAS_MAX_CONN MIN(CONFIG_BT_MAX_CONN, CONFIG_BT_MAX_PAIRED)
 
+#define BIT_CHANGED(_new_value, _old_value) ((_new_value) ^ (_old_value))
+#define FEATURE_DEVICE_TYPE_UNCHANGED(_new_type) \
+	!(BIT_CHANGED((_new_type) & BT_HAS_FEAT_HEARING_AID_TYPE_LSB, \
+			      (has.features & BT_HAS_FEAT_HEARING_AID_TYPE_LSB))) && \
+	!(BIT_CHANGED((_new_type) & BT_HAS_FEAT_HEARING_AID_TYPE_MSB, \
+			      (has.features & BT_HAS_FEAT_HEARING_AID_TYPE_MSB)))
+#define FEATURE_SYNC_SUPPORT_UNCHANGED(_new_value) \
+	!BIT_CHANGED(_new_value, ((has.features & BT_HAS_FEAT_PRESET_SYNC_SUPP) != 0 ? 1 : 0))
```

Comment:
Here you convert the return value of `BIT_CHANGED` to either 1 or 0. 

I suggest to either make `BIT_CHANGED` return a boolean value, or return `true` and `false` here as well as in `FEATURE_DEVICE_TYPE_UNCHANGED`